back/qc_api/lib/backtest/backtester.py

```diff
@@ -0,0 +1,157 @@
+            "list": []
+        }
+        self.wallet = Wallet(budget=self.budget)
+
+    def set_date(self, date: datetime.date) -> None:
+        """Sets self.today"""
+        self.today = date
+
+    def set_universe(self) -> None:
+        """Brings today (in context of backtesting)'s stock data from StockData DB table into memory"""
+        self.universe = pd.DataFrame(list(StockData.objects.filter(date=self.today).values()),
+                                     columns=stock_data_columns)
+
+    def update_wallet(self):
+        self.wallet.update_coins(universe_today=self.universe)
+
+    def make_sell_order(self):
+        sell_candidates = self.wallet.get_coins()
+        if len(sell_candidates) == 0:
+            pass
+        else:
+            chosen_stocks = []
+            exec_dict = {}
+            scope = dict(locals(), **globals())
+            exec(self.snippet_sell, scope, exec_dict)
+            if exec_dict.get("chosen_stocks"):
+                chosen_stocks = exec_dict.get("chosen_stocks")
+            self.make_amount_list(opt="sell", chosen_stocks=chosen_stocks)
+            for index, stock_tuple in enumerate(self.sell_amount_list):
+                self.wallet.sell_coin(stock=stock_tuple[0], amount=stock_tuple[1], time=self.today)
+
+    def make_buy_order(self):
+        if len(self.scope) == 0:
+            pass
+        else:
+            scope = self.scope
+            chosen_stocks = []
+            exec_dict = {}
+            _scope = dict(locals(), **globals())
+            exec(self.snippet_buy, _scope, exec_dict)
+            self.make_amount_list(opt="buy", chosen_stocks=chosen_stocks)
+            for index, stock_tuple in enumerate(self.buy_amount_list):
+                try:
+                    self.wallet.purchase_coin(stock=stock_tuple[0], amount=stock_tuple[1], time=self.today)
+                except IndexError:
+                    pass
+
+    def make_scope(self):
+        universe = self.universe
+        exec_dict = {}
+        _scope = dict(locals(), **globals())
+        exec(self.snippet_scope, _scope, exec_dict)
+        self.scope = exec_dict.get("scope")
+
+    def make_amount_list(self, opt, chosen_stocks):
+        buy_amount_list = []
+        sell_amount_list = []
+        exec_dict = {}
+        scope = dict(locals(), **globals())
+        if opt == "buy":
+            exec(self.snippet_amount, scope, exec_dict)
+            self.buy_amount_list = buy_amount_list
+        else:
+            exec(self.snippet_amount, scope, exec_dict)
+            self.sell_amount_list = sell_amount_list
+
+    def make_daily_report(self):
+        profit = self.wallet.get_profit()
+        self.report.get("transaction_log").append(self.wallet.get_transaction_log(self.today))
+        self.report.get("daily_profit").append({"date": str(self.today), "profit": profit})
+        self.track_max_min(profit=profit)
+
+    def track_max_min(self, profit):
```

Comment:
Please specify the rules used in here. What does 'eq', 'gt', 'lt' mean and how are they used?

Reply:
Method docstring and return value